compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -11262,6 +11272,12 @@ OMR::Z::TreeEvaluator::loadaddrEvaluator(TR::Node * node, TR::CodeGenerator * cg
                                                      (uintptr_t) node->getSymbol()->getStaticSymbol()->getStaticAddress(),
                                                      TR_BodyInfoAddress, NULL, NULL, NULL);
                }
+            else if (cg->comp()->compileRelocatableCode() && node->getSymbol()->isCatchBlockCounter())
```

Comment:
It's better to replace `compileRelocatableCode()` with something like `needRelocationsForBodyInfoData()`